File: LeetCode/Java/771-tree-宝石与石头.java
/*
 * @lc app=leetcode.cn id=771 lang=java
 *
 * [771] 宝石与石头
 */

// @lc code=start
class Solution {
    public int numJewelsInStones(String J, String S) {
        boolean[] hash = new boolean[256];
        for (char c : J.toCharArray())
            hash[c] = true;
        int res = 0;
        for (char c : S.toCharArray())
            if (hash[c])
                ++res;
        return res;
    }
}
// @lc code=end

File: LeetCode/C/771-tree-宝石与石头.cpp
/*
 * @lc app=leetcode.cn id=771 lang=cpp
 *
 * [771] 宝石与石头
 */
#include "main.h"

// @lc code=start
class Solution {
  public:
    int numJewelsInStones(string J, string S) {
        bool v[128] = {false};
        for (char c : J)
            v[c] = true;
        int res = 0;
        for (char c : S)
            if (v[c])
                ++res;
        return res;
    }
};
// @lc code=end
